src/Paprika/Chain/Blockchain.cs

```diff
@@ -417,6 +424,8 @@ private void Set(in Key key, in ReadOnlySpan<byte> payload)
             map.TrySet(key, payload);
         }
 
+        public IKeyEnumerator GetEnumerator() => throw new NotImplementedException("Not implemented yet");
```

Comment:
If `Block` is responsible for keeping track of the keys used, could we face performance issues?